### DISCUSSION TITLE
how to use `carapace-parse`?

### DISCUSSION THREAD

**Discussion by minhtrancccp** [carapace-sh/carapace-bin#2489] [Q&A] [answered]
i would like to generate a spec using the `--help` output automatically just like with [`dim-an/cod`](https://github.com/dim-an/cod), and [`carapace-parse`](https://github.com/carapace-sh/carapace/blob/master/docs/src/carapace/standalone/carapace-parse.md) seems to be fit for my case. it would be helpful if anyone can provide an instruction for building a binary of the parsing tool, as i'm not familiar with building go projects. many thanks.

**Comment by rsteube** [comment, marked as answer]
Ah yes, that one is still quite crude.

Running `go build` in [cmd/carapace-parse](https://github.com/carapace-sh/carapace-bin/tree/master/cmd/carapace-parse) is enough.

At the moment it only outputs go code so it won't help you. but converting it for spec support is possible.
The core of it is just this [regex](https://github.com/carapace-sh/carapace-bin/blob/master/cmd/carapace-parse/cmd/root.go#L147).

I don't have good documentation for it at the moment: https://github.com/carapace-sh/carapace-bin/pull/2415#issuecomment-2181556099
